server/src/main/java/org/kaazing/gateway/server/update/check/UpdateCheck.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.server.update.check;
+
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.kaazing.gateway.server.impl.VersionUtils.getGatewayProductEdition;
+import static org.kaazing.gateway.server.impl.VersionUtils.getGatewayProductTitle;
+import static org.kaazing.gateway.server.impl.VersionUtils.getGatewayProductVersionPatch;
+import static org.kaazing.gateway.server.update.check.GatewayVersion.parseGatewayVersion;
+
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+
+import javax.annotation.Resource;
+
+import org.kaazing.gateway.util.InternalSystemProperty;
+import org.kaazing.gateway.util.scheduler.SchedulerProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates and manages periodic checks to see if the gateway has updates
+ */
+public class UpdateCheck {
```

Comment:
This class should be the one extending from `GatewayObserverFactorySpiPrototype`, and override the `initingGateway` method.